src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -48,15 +52,30 @@ public void Init(IProviderConfiguration providerConfig, string providerName, Log
 
             logger = log;
             serviceProvider = svcProvider;
+            receivers = new ConcurrentDictionary<QueueId, EventHubAdapterReceiver>();
+
             adapterConfig = new EventHubStreamProviderConfig(providerName);
             adapterConfig.PopulateFromProviderConfig(providerConfig);
-
             hubSettings = adapterConfig.GetEventHubSettings(providerConfig, serviceProvider);
-            checkpointSettings = adapterConfig.GetCheckpointSettings(providerConfig, serviceProvider);
-
-            receivers = new ConcurrentDictionary<QueueId, EventHubAdapterReceiver>();
             client = EventHubClient.CreateFromConnectionString(hubSettings.ConnectionString, hubSettings.Path);
-            bufferPool = new FixedSizeObjectPool<FixedSizeBuffer>(adapterConfig.CacheSizeMb, pool => new FixedSizeBuffer(1 << 20, pool));
+
+            if (CheckpointFactory == null)
+            {
+                checkpointSettings = adapterConfig.GetCheckpointSettings(providerConfig, serviceProvider);
+                CheckpointFactory = partition => EventHubCheckpoint.Create(checkpointSettings, adapterConfig.StreamProviderName, partition);
```

Comment:
This change enables what you described, but was not the core intent.  The main goal was to allow application developers to provide their own checkpoint logic so they'd not be tied to the default azure table storage implementation.